src/flashlight.c

```diff
@@ -82,9 +94,9 @@ toggle_flashlight_action(GAction *action,
       if (needs_enable && fd2 != NULL)
         fprintf(fd2, "0");
       else
-        fprintf(fd1, QCOM_DISABLE);
+        fprintf(fd1, (is_qcom_torch ? QCOM_DISABLE : MTK_DISABLE));
```

Comment:
Then based on vendor variable execution path can be split into toggle_flashlight_action_qcom/toggle_flashlight_action_mtk. This may lead to a bit of code duplication, but should be much easier to read/understand what is going on.